packages/widget/src/components/overlay-component/overlay-component.ts

```diff
@@ -0,0 +1,32 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { when } from 'lit/directives/when.js';
+import { styles } from './styles';
+
+@customElement('sygma-overlay-component')
+export class OverlayComponent extends LitElement {
+  static styles = styles;
+
+  @property({ type: Boolean })
+  isLoading = false;
+
+  render(): HTMLTemplateResult {
+    return when(
+      this.isLoading,
```

Comment:
IMHO it's much better not to render/mount component, instead of passing boolean prop which says if component should render anything.